### PR TITLE
support Error codes 

### PR DESCRIPTION
**Pull request by barbara79**

### CODE REVIEW

File: types/error/EditPreviewFailedError.raml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#%RAML 1.0 DataType
+(package): Error
+type: ErrorObject
+displayName: EditPreviewFailedError
+discriminatorValue: EditPreviewFailed
+properties:
+  result:
+    type: OrderEditPreviewFailure
```
Comment (barbara79) [inline types/error/EditPreviewFailedError.raml:8]: Is it the exact type?

File: types/error/MaxResourceLimitExceededError.raml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#%RAML 1.0 DataType
+(package): Error
+type: ErrorObject
+displayName: MaxResourceLimitExceededError
+discriminatorValue: MaxResourceLimitExceeded
+properties:
+  exceededResource:
+    type: ReferenceTypeId
```
Comment (barbara79) [inline types/error/MaxResourceLimitExceededError.raml:8]: in the docs is pointing to the ReferenceType so is it correct to put as a type the Id? it doesn't work with ReferenceType

File: types/error/SemanticErrorError.raml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#%RAML 1.0 DataType
+(package): Error
+type: ErrorObject
+displayName: SemanticErrorError
```
Comment (barbara79) [inline types/error/SemanticErrorError.raml:4]: for the convention that we have in this repository I had to write "...ErrorError", not sure if it should be useful like this

File: types/error/SyntaxErrorError.raml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#%RAML 1.0 DataType
+(package): Error
+type: ErrorObject
+displayName: SyntaxErrorError
```
Comment (barbara79) [inline types/error/SyntaxErrorError.raml:4]: same comment as "SemanticErrorError"

File: types/error/GeneralError.raml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#%RAML 1.0 DataType
+(package): Error
+type: ErrorObject
+displayName: GeneralError
+discriminatorValue: General
```
Comment (barbara79) [inline types/error/GeneralError.raml:5]: not sure if it's correct to add this error but it's in the documentation

File: types/error/MaxResourceLimitExceededError.raml (hunk 2 above)
Comment (jenschude) [reply types/error/MaxResourceLimitExceededError.raml:8]: It's only named differently in the API ref, so it's fine

File: types/error/EditPreviewFailedError.raml (hunk 1 above)
Comment (jenschude) [reply types/error/EditPreviewFailedError.raml:8]: yes

File: types/error/SemanticErrorError.raml (hunk 3 above)
Comment (jenschude) [reply types/error/SemanticErrorError.raml:4]: looks ugly but it's still correct

File: types/error/GeneralError.raml (hunk 5 above)
Comment (jenschude) [reply types/error/GeneralError.raml:5]: it's a valid error

**Review by jenschude — APPROVED** [review]

**Pull request merged**